flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -88,13 +347,280 @@ def add_wind_forcing(
+
+        # HydroMT function: create storage volume
+        self._model.setup_storage_volume(
+            storage_locs=gdf_green_infra, volume=volume, height=height, merge=True
+        )
+
+    def _add_measure_pump(self, pump: Pump):
+        """Add pump to sfincs model.
+
+        Parameters
+        ----------
+        pump : PumpModel
+            pump information
+        """
+        polygon_file = Database().input_path.joinpath(pump.attrs.polygon_file)
+        # HydroMT function: get geodataframe from filename
+        gdf_pump = self._model.data_catalog.get_geodataframe(
+            polygon_file, geom=self._model.region, crs=self._model.crs
+        )
+
+        # HydroMT function: create floodwall
+        self._model.setup_drainage_structures(
+            structures=gdf_pump,
+            stype="pump",
+            discharge=pump.attrs.discharge.convert(UnitTypesDischarge("m3/s")),
+            merge=True,
         )
 
-    def add_wind_forcing_from_grid(self, ds: xr.DataArray):
+    ### PROJECTIONS HELPERS - not needed at the moment ###
+
+    ##### OLD CODE BELOW #####
+    # @Gundula: I have renamed some the functions and added comments to some that explain when they were called in the original hazard.py just to keep track.
+    # Im not super familiar with hydromt_sfincs and its functions, so I have not changed the logic of the functions, only the names and comments.
+    # It could be the case that some of the functions in here are not needed anymore, but we can always remove them later if needed.
+
+    # I have also added a '_' to the start of all functions I see as private.
+    # Public functions can be called from outside of the class, all private functions should not be called from outside of the class.
+    # (even though python does not care and will allow it anyways, we can still follow the convention just to make it clear)
+
+    ### SFINCS SETTERS ###
+    def _set_timing(self, event: EventModel):
+        """Set model reference times based on event time series."""
+        # Get start and end time of event
+        tstart = event.time.start_time
+        tstop = event.time.end_time
+
+        # Update timing of the model
+        self._model.set_config("tref", tstart)
+        self._model.set_config("tstart", tstart)
+        self._model.set_config("tstop", tstop)
+
+    def _set_config_spw(self, spw_name: str):
+        self._model.set_config("spwfile", spw_name)
+
+    def _turn_off_bnd_press_correction(self):
+        self._model.set_config("pavbnd", -9999)
+
+    ### ADD TO SFINCS ###
+    # @Gundula functions starting with `add` should not blindly overwrite data in sfincs, but read data from the model, add something to it, then set the model again.
+    # This is to ensure that we do not overwrite any existing data in the model.
+    # (maybe this is what hydromt_sfincs already does, but just checking with you)
+    def _add_obs_points(self):
```

Comment:
the base model could have lots of observation points for validation during the model set up that are not needed anymore later. I think we should overwrite them with the observation points defined in the site.toml

Reply:
Sounds good to me. I think if we add some get and set functions, this will be quite easy to do. If we then want to add to the model, we first cal get_x, modify the data, then call set_x. Making it very clear whats happening